### DISCUSSION TITLE
$range helper

### DISCUSSION THREAD

**Discussion by KevinBatdorf** [alpine-collective/alpine-magic-helpers#30] [Ideas] [answered]
Putting this here to discuss the idea of a `$range` helper

This issue is closed now, but for Alpine 3, could it be changed to be `i in $range(10)`? I think making a magic `$range()` function that works like Python's range ($range(2) = [1, 2], $range(2, 4)= [2, 3], $range(0, 4, 2) = [0, 2]) would be simpler to implement and simpler to understand, as well as allowing more flexibility (e.g. `$range({ from: 1, to: 2, inclusive: true})` = [1, 2]).

_Originally posted by @carlmjohnson in https://github.com/alpinejs/alpine/pull/748#issuecomment-723477685_

**Comment by SimoTod** [comment]
+1 I think it could be a good candidate for the library

**Reply by HugoDF** [reply]
Agree with Simone 👍

Should be pretty straightforward to implement + will give extra flexibility from the built-in syntax.

**Comment by KevinBatdorf** [comment, marked as answer]
Added it here if someone can take a look https://github.com/alpine-collective/alpine-magic-helpers/pull/31

If it looks good, we could release `$scroll` and `$range` today and just hold off on `$undo` until Caleb releases a new version with the clone changes.
